README.md

```
@@ -1,14 +1,26 @@
 
-My plan is to first implement a solution that will enable the visual tool to do the mapping in the design workbench and save that in the engine. Then I will bring the engine into the instrumenter and use the mapping to instrument the program so it produces a trace that can be transforme into the behavior of the design. Then the rest of the design feedback loop will take over.
+This program will be called from the node server and it will provide the necessary metadata to the UI to enable the mapping. It will then be called by the server again when it instruments and executes the code. The resulting execution trace will be automatically debugged by the engine. 
 
-This will be written in python so that I can leverage its AST library and the server will invoke this program as needed. Eventually this will be extended for other languages and the same process will repeat.
+This process will be repeated for multiple languages and it will also be extended to instrument systems. The instrumenter will use CLP logging libraries and the mapping will be extended to include domain specific knowledge about the data to apply domain specific compression.
 
-[adli]: https://github.com/vishalpalaniappan/asp-adli-python
+## Usage
+
+To produce mapping from a python program:
+```
```

markdownlint-cli2 (0.22.0)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)